numpy/core/src/_simd/_simd.dispatch.c.src

```diff
@@ -0,0 +1,367 @@
+/*@targets $werror #simd_test*/
+#include "_simd.h"
+#include "_simd_inc.h"
+
+#if NPY_SIMD
+#include "_simd_inc_data.h"
+#include "_simd_inc_convert.h"
+#include "_simd_inc_vector.h"
+#include "_simd_inc_arg.h"
+#include "_simd_inc_easyintrin.h"
+#endif
+
+/*************************************************************************
+ * Defining NPYV intrinsics as module functions
+ *************************************************************************/
+#if NPY_SIMD
+/**begin repeat
+ * #sfx       = u8, s8, u16, s16, u32, s32, u64, s64, f32, f64#
+ * #bsfx      = b8, b8, b16, b16, b32, b32, b64, b64, b32, b64#
+ * #simd_sup  = 1,  1,  1,   1,   1,   1,   1,   1,   1,   NPY_SIMD_F64#
+ * #sat_sup   = 1,  1,  1,   1,   0,   0,   0,   0,   0,   0#
+ * #mul_sup   = 1,  1,  1,   1,   1,   1,   0,   0,   1,   1#
+ * #div_sup   = 0,  0,  0,   0,   0,   0,   0,   0,   1,   1#
+ * #shl_imm   = 0,  0,  15,  15,  31,  31,  63,  63,  0,   0#
+ * #shr_imm   = 0,  0,  16,  16,  32,  32,  64,  64,  0,   0#
+ */
+#if @simd_sup@
+/***************************
+ * Memory
+ ***************************/
+/**begin repeat1
+ * # intrin = load, loada, loads, loadl#
+ */
+SIMD_IMPL_INTRIN_1(@intrin@_@sfx@, v@sfx@, q@sfx@)
+/**end repeat1**/
+/**begin repeat1
+ * # intrin = store, storea, stores, storel, storeh#
+ */
+// special definition due to the nature of @intrin@
+static PyObject *
+simd__intrin_@intrin@_@sfx@(PyObject* NPY_UNUSED(self), PyObject *args)
+{
+    simd_arg req_args[] = {
+        {.type = simd_data_q@sfx@},
+        {.type = simd_data_v@sfx@},
+    };
+    if (!simd_arg_parse(args, req_args, 2)) {
```

Comment:
I'd recommend using the `PyArg_ParseTuple` machinery here. You'd write a `SimdArgConverter` function (returning `1` on error, irritatingly), and then this would become
```suggestion
    if (!PyArg_ParseTuple("intrin_@intrin@_@sfx@:O&O&", args, &req_args[0], &req_args[1])) {
```

Reply:
simd_arg_parse is faster, easy to use, and friendly with GDB. PyArg_ParseTuple is kinda annoying and too slow NumPy should avoid using it at all.

Reply:
The point in this PR is not to write a super-fast arg-parsing library. `PyArg_Parse` is the obvious way to parse arguments, and it provides better error messages too.

The function you'd need to write is as simple as
```python
static int
SimdArgConverter(PyObject *obj, simd_arg *a)
{
    if (obj != NULL) {
        a->obj = obj
        a->data = simd_obj2data(a->obj, a->type);
        if (PyErr_Occurred()) {
            return Py_CLEANUP_SUPPORTED;
        }
        return 1;
    }
    else {
        /* this is the cleanup path caused by later arguments failing */
        simd_obj2data_clear(a);  // your code did not do this, and leaks memory on error
    }
}
```

Reply:
> The point in this PR is not to write a super-fast arg-parsing library.

I'm not sure about the future of this module, maybe later we can add other data types to handle big-chunks of memory,
but I guess you're right _SIMD module should be only for testing NPYV intrinsics.

> The function you'd need to write is as simple as the following and is about 40 lines shorter than simd_arg_parse

I will give it a try and let us see.

Reply:
You are right that `PyArg_ParseTuple` and friends are slow - which is why Python has switched to using "argument clinic" to generate C code to parse arguments instead.

Reply:
please give me your final call. Are you sure about using `PyArg_ParseTuple`?  or Should I just keep `simd_arg_parse` after removes pytuple check?

Reply:
I think we should stay with `PyArg_ParseTuple` here. We can always come back to this PR if it becomes a performance bottleneck

Reply:
please feel free to drop the last patch in case if you changed your mind, IMHO iterate args tuple directly is far away better.